### DISCUSSION TITLE
Berry console length limit

### DISCUSSION THREAD

**Discussion by msetina** [arendst/Tasmota#19718] [Q&A] [answered]
What is the limit for Berry console?
I have a class that has 337 lines and the Berry console discracefully fails.
It is hard to check, but I think that error reply is replaced by the actual HTML of the berry console page.

**Comment by s-hadinger** [comment]
It depends on available memory. Without PSRAM it fails indeed past a certain size. I'm not sure how to avoid this. If you find a way, please suggest it.

**Reply by msetina** [reply]
Ok. Can Berry console show memory available to it?
If the error showed that it ran out of RAM not just returning HTML in console instead pushing it to client browser, it would help.

**Reply by s-hadinger** [reply]
I'm sorry but it's not in my priority list, so I would be grateful if someone is willing to work on it

**Reply by sfromis** [reply, marked as answer]
For more than small files, I'd recommend to not paste it all into the Berry REPL console, but have it in the file system, and load it from there. Basically, I do not "like" working when a vertical scroll bar appears in the input text area extended to what fits on the screen.

If you use a custom build with `USE_BERRY_DEBUG` you get a red "Restart Berry VM (for devs only)" button on the REPL web page, meaning that if you want to start afresh with a test case pasted to the input area, you should be able to clean out first, reducing the risk of running out of RAM, and getting a more predictable "cutoff point". Also reduces risk of errors based on things you previously did still being available in memory.

Such a build is recommended anyway, as you get more debug info like line numbers for many exceptions. OTOH, debug enabled does cost some RAM. Having PSRAM available when developing can also be recommended a lot.

**Reply by msetina** [reply]
Thanks. I went the file on filesystem way and was succesfull and got usefull scenario. It would be nice to have a debug build in standard builds, but if resources are constrained, I understand. 
For now I am trying to get by with what is available as standard. Will see how far it gets me.
